rex/utilities/utilities.py

```diff
@@ -731,3 +732,35 @@ def slice_sites(shape, chunks, sites=None, chunks_per_slice=5):
         raise TypeError(msg)
 
     return slices
+
+
+def res_dist_threshold(lat_lons, tree=None, margin=1.05):
+    """
+    Estimate the distance between resource points. Calculated as half of the
+    diagonal between closest resource points, with desired extra margin
+
+    Parameters
+    ----------
+    lat_lons : ndarray
+        n x 2 array of resource points coordinates (lat, lon)
+    tree : cKDTree, optional
+        Pre-build cKDTree of resource lat, lon coordintes. If None, build the
+        cKDTree from scratch, by default None
+    margin : float, optional
+        Extra margin to multiply times the computed max distance between
+        neighboring resource points, by default 1.05
+
+    Returns
+    -------
+    float
+        Estimate the distance between resource points. Calculated as half of
+        the diagonal between closest resource points, with desired extra margin
+    """
+    if tree is None:
+        # pylint: disable=not-callable
+        tree = cKDTree(lat_lons)
+
+    dists = tree.query(lat_lons, k=2)[0][:, 1]
+    dists = dists[(dists != 0)]
+
+    return margin * (2 ** 0.5) * (dists.min() / 2)
```

Comment:
Do you want to do `dists.min()` here? As opposed to mean, median, or max? The docstring in margin says it uses the max distance between neighboring resource points. I think maybe the best option is to use the 95th percentile or something similar?

Reply:
I think we should use max then. Anything less will incorrectly screen out tree query results in portions of the grid that are stretched... right?

Reply:
agreed, will update